snow/networking/sender/sender.go

```diff
@@ -34,27 +34,45 @@
 
 // GetAcceptedFrontier ...
 func (s *Sender) GetAcceptedFrontier(validatorIDs ids.ShortSet, requestID uint32) {
-	currentDeadline := time.Time{}
+	// TODO this timeout duration won't exactly match the one that gets registered.
+	// I think that's OK, but double check.
+	timeoutDuration := s.timeouts.TimeoutDuration()
+	deadline := time.Now().Add(timeoutDuration)
+
+	// A GetAcceptedFrontier to myself will always fail.
+	if validatorIDs.Contains(s.ctx.NodeID) {
+		validatorIDs.Remove(s.ctx.NodeID)
+		go s.router.GetAcceptedFrontierFailed(s.ctx.NodeID, s.ctx.ChainID, requestID)
+	}
+
+	// Some of the validators in [validatorIDs] may be benched. That is, they've been unresponsive
+	// so we don't even bother sending messages to them. We just have them immediately fail.
 	for validatorID := range validatorIDs {
+		if s.timeouts.IsBenched(validatorID, s.ctx.ChainID) {
+			validatorIDs.Remove(validatorID)
+			// Immediately register a failure. Do so asynchronously to avoid deadlock.
+			go s.router.GetAcceptedFrontierFailed(s.ctx.NodeID, s.ctx.ChainID, requestID)
```

Comment:
Is there value in some way marking that these failures are due to the validator being benched ?  And separating these from other failures ? 

At least from a metrics/logs/visibility perspective ?

Reply:
Added a metric for that :+1: